src/Config/Converters/NestedMutationOptionsConverter.cs

```diff
@@ -0,0 +1,88 @@
+using System.Text.Json.Serialization;
+using Azure.DataApiBuilder.Config.ObjectModel;
+
+namespace Azure.DataApiBuilder.Config.Converters
+{
+    /// <summary>
+    /// Converter for the nested mutation options.
+    /// </summary>
+    internal class NestedMutationOptionsConverter : JsonConverter<NestedMutationOptions>
+    {
+
+        private readonly NestedCreateOptionsConverter _nestedCreateOptionsConverter;
+
+        public NestedMutationOptionsConverter(JsonSerializerOptions options)
+        {
+            _nestedCreateOptionsConverter = options.GetConverter(typeof(NestedCreateOptions)) as NestedCreateOptionsConverter ??
+                                            throw new JsonException("Failed to get nested create options converter");
+        }
+
+        /// <inheritdoc/>
+        public override NestedMutationOptions? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            if (reader.TokenType is JsonTokenType.StartObject)
+            {
+                NestedMutationOptions? nestedMutationOptions = null;
+
+                while (reader.Read())
+                {
+                    if (reader.TokenType == JsonTokenType.EndObject)
+                    {
+                        break;
+                    }
+
+                    string? propertyName = reader.GetString();
+                    switch (propertyName)
+                    {
+                        case "create":
+                            reader.Read();
+                            NestedCreateOptions? nestedCreateOptions = _nestedCreateOptionsConverter.Read(ref reader, typeToConvert, options);
+                            if (nestedCreateOptions is not null)
+                            {
+                                nestedMutationOptions = new(nestedCreateOptions);
+                            }
+
+                            break;
+
+                        default:
+                            throw new JsonException($"Unexpected property {propertyName}");
+                    }
+                }
+
+                return nestedMutationOptions;
+            }
+
+            throw new JsonException("Failed to read the GraphQL Nested Mutation options");
+        }
+
+        /// <inheritdoc/>
+        public override void Write(Utf8JsonWriter writer, NestedMutationOptions? value, JsonSerializerOptions options)
+        {
+            // If the nested mutation options is null, it is not written to the config file.
+            if (value is null)
+            {
+                return;
+            }
+
+            writer.WritePropertyName("nested-mutations");
+
+            writer.WriteStartObject();
+
+            if (value.NestedCreateOptions is not null)
+            {
+                _nestedCreateOptionsConverter.Write(writer, value.NestedCreateOptions, options);
+            }
+
```

Comment:
if `(value.NestedCreateOptions` is null, then i'd argue that we wouldn't want to write the nested mutations property right? that would be invalid config to provide:
1.  `"nested-mutations": {}` 
2.  `"nested-mutations": null`

Reply:
In L71, I am performing a `null` check against `value` which is of type `NestedMutationOptions`. If it is `null`, then, nothing will be written to the config file. So, the config file created using CLI will not have null or empty values for `nested-mutations` property. 


Right now, `NestedMutationOptions` will be non-null, only when `NestedCreateOptions` is also not null. 

However, in the future, if we introduce `NestedUpdateOptions`, then it is possible that `NestedMutationOptions` is non-null but `NestedCreateOptions` is null. But, that should also get taken care of as a `null` check is performed on L80 (before writing the fields related to nested create out to the config file).

Reply:
I ack that configs such as `"nested-mutations": {}` and `"nested-mutations": null` are possible as a result of users manually editing the config file instead of using CLI. I've handled these scenarios when reading the config file and added tests to validate the same.